app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java

```diff
@@ -136,17 +141,97 @@ public void onMapReady(GoogleMap googleMap) {
       CommonTools.showSnackbar(requireView(), e.getMessage());
       return;
     }
+    if (getArguments() != null) {
+      intentType = getArguments().getInt(LOCATION_ARGUMENT_KEY);
+      latitudeFromChat = getArguments().getString("LATITUDE_ARGS");
+      longitudeFromChat = getArguments().getString("LONGITUDE_ARGS");
+    }
+
+    if (intentType != 0) { // if intent is to edit, request, share, or observe
+      setLimitedView();
+    }
     try {
-      setupNearbyFavorsListener();
-      setupFocusedFavorListen();
+      // only when the app is firstly opened, center on my location,
+      // otherwise just return where I left before
+      if (!getViewModel().isShowObservedFavor() && intentType != 0) //
+      {
+        boolean isMarkerEditable = intentType != OBSERVE_LOCATION;
+        Marker marker = drawMarkerAndFocusOnLocation(isMarkerEditable);
+        doneButton.setOnClickListener(
+            v -> {
+              if (intentType == SHARE_LOCATION) sendLocationToChat(marker);
```

Comment:
I notice if I drag a marker and share a new location,  the new location is actually the old one.
You can replace with this line code to fix it, 
`if (intentType == SHARE_LOCATION) sendLocationToChat(newMarkers.get(0));`
and another change at line 269

Reply:
I cannot leave comments at line 269, so just leave it here: 
 `  @Override
    public void onMarkerDragEnd(Marker marker) {
      newMarkers.set(0, marker);
    }`

Reply:
Great! thanks for pointing that out